packages/api/src/server/analytics/service/analytics.service.ts

```diff
@@ -211,57 +211,79 @@ export class AnalyticsService {
-    const totalTimeForStandardDeployment = Number(ANALYTICS.averageTimeToDeploy) * response.count;
+    const totalTimeForStandardDeployment = Number(averageDeploymentTimeInSecs || ANALYTICS.averageTimeToDeploy) * response.count;
     const timeSavedInSec = totalTimeForStandardDeployment - response.totalTime;
     const timeSavedInHours = Math.round(timeSavedInSec / AnalyticsService.seconds / AnalyticsService.minutes);
     return { timeSavedInHours, averageTime: response.averageTime, deploymentCount: response.count };
   }
 
-  async getDeveloperMetrics(month: number, cluster: string, type: string): Promise<Object> {
+  async getDeveloperMetrics(month: number, cluster: string, type: string, mode: string, averageDeploymentTimeInSecs: string): Promise<Object> {
     const monthlyDateFrame = await this.analyticsFactory.buildMonthlyDateFrame(month || 1);
+    const spashipAdmin = 'spaship-admin';
     let overallCostSaved = 0;
-    const { averageDevelopmentHours } = ANALYTICS;
+    const averageDevelopmentHours = Number(averageDeploymentTimeInSecs || ANALYTICS.averageDevelopmentHours);
+    const { developerHourlyRate } = ANALYTICS;
     const response = [];
     for (const tmpMonth of monthlyDateFrame) {
-      const monthlyAnalytics = await this.getAverageDeploymentTime('', 'NA', AnalyticsService.days, tmpMonth, cluster, type);
-      const spashipAverageTimeInSecs = monthlyAnalytics.averageTime;
-      const averageSavedTimeInSecs = Number(averageDevelopmentHours) - monthlyAnalytics.averageTime;
-      const totalWorkingHours = Number(ANALYTICS.workingDays) * Number(ANALYTICS.workingHours);
-      const totalDeploymentCount = monthlyAnalytics.count;
-      const totalDeploymentHours = parseFloat(
-        ((monthlyAnalytics.averageTime * totalDeploymentCount) / AnalyticsService.seconds / AnalyticsService.minutes).toFixed(2)
-      );
-      const frequencyOfDeployment = parseFloat((totalDeploymentCount / totalWorkingHours).toFixed(2));
-      const { developerHourlyRate } = ANALYTICS;
-      const costSavingPerHour = parseFloat(
-        (
-          (averageSavedTimeInSecs / AnalyticsService.seconds / AnalyticsService.minutes) *
-          frequencyOfDeployment *
-          Number(developerHourlyRate)
-        ).toFixed(2)
-      );
-      const totalCostSaved = parseFloat((costSavingPerHour * totalDeploymentHours).toFixed(2));
-      if (totalCostSaved) overallCostSaved += totalCostSaved;
-      response.push({
-        ...tmpMonth,
-        averageSavedTimeInSecs,
-        spashipAverageTimeInSecs,
-        totalWorkingHours,
-        totalDeploymentCount,
-        totalDeploymentHours,
-        frequencyOfDeployment,
-        developerHourlyRate,
-        costSavingPerHour,
-        totalCostSaved
-      });
+      let frequencyOfDeployment;
+      let costSavingPerHour;
+      let totalCostSaved;
+      try {
+        const monthlyAnalytics = await this.getAverageDeploymentTime('', 'NA', AnalyticsService.days, tmpMonth, cluster, type);
```

Comment:
Hi @arkaprovob , here we're collecting the analytics for every month, if for any reason this ( `getAverageDeploymentTime` ) month throws an Error. We're handling it with the exception, and skipping that month.